pkg/lvm/lvm_util.go

```diff
@@ -254,6 +256,71 @@ func ResizeLVMVolume(vol *apis.LVMVolume, resizefs bool) error {
 	return err
 }
 
+// ResizeBTRFSVolume resizes the LVM volume as well as expand btrfs filesystem
+// Step1: Expanding LVM Volume (lvextend lvmvg -n <volume_name> -L size)
+// Step2: Expanding btrfs filesystem (btrfs filesystem resize max <mount_path>)
+func ResizeBTRFSVolume(vol *apis.LVMVolume, mountPath string) error {
+
+	desiredVolSize, err := strconv.ParseUint(vol.Spec.Capacity, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	curVolSize, err := getLVSize(vol)
+	if err != nil {
+		return err
+	}
+
+	// Expand LVM volume only when desired volume size is greater
+	// than current volume size. This will handles a case where
+	// LVM expansion is succeeded but not the FS expansion in earlier
+	// requests
+	if desiredVolSize > curVolSize {
+		err := ResizeLVMVolume(vol, false)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Expand btrfs filesystem
+	err = btrfs.ResizeBTRFS(mountPath)
```

Comment:
what happens if we have not resizeed the volume but calling btrfs fs resize, or resizing the btrfs fs two or more times, will it fail?

Reply:
It will not throw any errors... Looks like it is acting same as `resize2fs`
```sh
[sai@centos-worker-1 ~]$ sudo btrfs filesystem resize max /mnt/store
Resize '/mnt/store' of 'max'
[sai@centos-worker-1 ~]$ echo $?
0
[sai@centos-worker-1 ~]$ sudo btrfs filesystem resize max /mnt/store
Resize '/mnt/store' of 'max'
[sai@centos-worker-1 ~]$ echo $?
0
[sai@centos-worker-1 ~]$ 

```